src/esm_runscripts/filedicts.py

```diff
@@ -527,6 +533,19 @@ def find_globbing_files(self, location : str) -> list:
 
         return glob_paths
 
+    def makedirs_in_name(self, name_type):
+        if "/" in self[f"name_in_{name_type}"]:
+            parent_path = self[f"absolute_path_in_{name_type}"].parent
+            if not parent_path.exists():
+                location = self.locations[name_type]
+                if location.exists():
+                    os.makedirs(parent_path)
+                else:
+                    raise FileNotFoundError(
+                        f"Unable to perform file operation. Path for ``{name_type}`` "
+                        f"({location}) does not exist!"
+                    )
+
```

Comment:
Only a comment: we should stay consistent with the parameter names. Here we have `name_type`, on the outside we have `target`. It would make it easier to read. Not a hard-must-have, but just an idea.

Reply:
I changed the name here, because we might in the future use this function for the source? I thought it was just more general.

Reply:
That makes sense 👍